src/Cli/dotnet/commands/dotnet-tool/update/ToolUpdateAllCommandParser.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using Microsoft.DotNet.Tools.Tool.Common;
+using Microsoft.DotNet.Tools.Tool.Update;
+using LocalizableStrings = Microsoft.DotNet.Tools.Tool.Update.LocalizableStrings;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class ToolUpdateAllCommandParser
+    {
+        public static readonly CliOption<bool> GlobalOption = ToolAppliedOption.GlobalOption;
+
+        public static readonly CliOption<string> ToolPathOption = ToolAppliedOption.ToolPathOption;
```

Comment:
Thanks for creating a new parser for the `--all` subcommand. It looks to me that some of the options might not make sense in `--all`, such as the `ToolPathOption` or `AddSourceOption`. We might need to reconsider the options here.

Reply:
Thank you! Will remove them shortly

Reply:
For this options, @dsplaisted suggests that it's possible to use the options in `AddCommandOptions` in `ToolInstallCommandParser` to avoid duplications in code. Essentially, most of the options in `dotnet tool update` should be applicable in `dotnet tool update` with `--all`. To avoid conflict, it's possible to add logic to the parser to check if there are any options which are conflicting.